packages/ember-glimmer/lib/helpers/each-in.js

```diff
@@ -0,0 +1,63 @@
+/**
+@module ember
+@submodule ember-templates
+*/
+
+import { UpdatableReference } from '../utils/references';
+import symbol from 'ember-metal/symbol';
+
+/**
+  The `{{each-in}}` helper loops over properties on an object. It is unbound,
+  in that new (or removed) properties added to the target object will not be
+  rendered.
+
+  For example, given a `user` object that looks like:
+
+  ```javascript
+  {
+    "name": "Shelly Sails",
+    "age": 42
+  }
+  ```
+
+  This template would display all properties on the `user`
+  object in a list:
+
+  ```handlebars
+  <ul>
+  {{#each-in user as |key value|}}
+    <li>{{key}}: {{value}}</li>
+  {{/each-in}}
+  </ul>
+  ```
+
+  Outputting their name and age.
+
+  @method each-in
+  @for Ember.Templates.helpers
+  @public
+  @since 2.1.0
+*/
+export default {
+  isInternalHelper: true,
+  toReference(args) {
+    return new EachInHelperReference(args.positional.at(0));
+  }
+};
+
+const EACH_IN_REFERENCE = symbol('EACH_IN');
+
+export function isEachIn(ref) {
+  return ref && ref[EACH_IN_REFERENCE];
+}
+
+class EachInHelperReference extends UpdatableReference {
+  constructor(sourceRef) {
+    super(sourceRef);
+    this[EACH_IN_REFERENCE] = true;
+  }
+
+  value() {
+    return super().value();
+  }
+}
```

Comment:
what about moving the isConst into the EachInReference#toReference? I believe this is what the get helper is doing.

If we do the wrapping using `Object.create` doesnt `Object.keys()` only return
```javascript
{
  [EACH_IN_REFERENCE]: true
}
```
since the key's of the underlying object reference aren't technically properties of the wrapped object, but of its prototype?